bears/generate_package.py

```diff
@@ -0,0 +1,83 @@
+import argparse
+import glob
+import os
+import shutil
+from string import Template
+import subprocess
+import sys
+
+from bears import VERSION
+from coalib.collecting.Importers import iimport_objects
+
+
+def touch(file_name):
+    """
+    Creates an empty file. An existing file remains untouched.
+
+    :param file_name: Name of the file.
+    """
+    if not os.path.exists(file_name):
+        open(file_name, 'w').close()
+
+
+def create_file_from_template(template_file, output_file, substitution_dict):
+    """
+    Creates a file from a template file, using a substitution dict.
+
+    :param template_file:     The template file.
+    :param output_file:       The file to be written.
+    :param substitution_dict: The dict from which the substitutions are taken.
+    """
+    with open(template_file) as fl:
+        template = fl.read()
+    template = Template(template).safe_substitute(**substitution_dict)
+
+    with open(output_file, 'w') as output_handle:
+        output_handle.write(template)
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Generates PyPi packages from bears.')
+    parser.add_argument('-u', '--upload', help='Upload the packages on PyPi',
+                        action='store_true')
+    args = parser.parse_args()
+
+    os.makedirs('upload', exist_ok=True)
+
+    for bear_file_name in glob.glob('**/*Bear.py'):
+        bear_name = os.path.splitext(os.path.basename(bear_file_name))[0]
+        bear_object = next(iimport_objects(bear_file_name, attributes='kind',
+                                           local=True))
+        upload_bear_folder = os.path.join('upload', bear_name, bear_name)
```

Comment:
Then all this stuff here is conceptually related, perhaps a `create_file_structure()` method?